### DISCUSSION TITLE
remove blue login screen

### DISCUSSION THREAD

**Discussion by blaizmannen** [neutrinolabs/xrdp#2429] [Q&A] [answered]
Is there any way to remove the default blue login screen with XRDP, to instead use the default screen for the OS? (i.e ubuntu login screen)
![image](https://user-images.githubusercontent.com/76086010/201883340-d29d501e-af00-4269-9901-4b0c9d396dbf.png)

This login screen occurs when:

1. When only inputting a username in Microsoft Remote Desktop
2. Getting the password wrong in Microsoft Remote Desktop

Otherwise if I save my username and password in Microsoft Remote Desktop, this blue login screen is skipped.

**Comment by matt335672** [comment]
There's a significant difference at the design level between the OS login screen and the xrdp login screen. There's a lot of history behind this.

The OS login screen is provided by one of several [display managers](https://en.wikipedia.org/wiki/X_display_manager) (lightdm, gdm, sddm, etc). Quite a bit of functionality may be provided other than just a login screen. Importantly reconnecting to an existing X session is not a supported use-case. Generally the lock screen is provided by a program other than the display manager, but gdm is an exception here.

The existing xrdp login screen can start a new session (with the help of sesman), connect to a session and also act as a proxy manager for existing VNC and RDP sessions on the network.

In other words, these are different programs trying to do different things, despite the superficial similarities.

**Reply by blaizmannen** [reply]
Thank you for the swift reply.
`Generally the lock screen is provided by a program other than the display manager, but gdm is an exception here.` you are referring to the xrdp lock screen?

We only need to be able to log in using a username and password, which the built-in ubuntu lock screen does just fine. I am wondering if we are able to completely bypass the xrdp lock screen. Sorry if I am asking the same question.

**Comment by matt335672** [comment, marked as answer]
There's no xrdp lock screen as such - there's just the login screen.

Depending on your Ubuntu version, the 'built-in ubuntu lock screen' is part of a much larger program - the display manager. The login screen isn't available on its own. So to get hold of it, we'd have to split it out from the display manager and maintain it ourselves which isn't a great idea (IMO).
